charts/spire/charts/spiffe-oidc-discovery-provider/templates/tests/test-keys.yaml

```diff
@@ -29,6 +29,27 @@ spec:
       volumeMounts:
         - name: data-volume
           mountPath: /data
+    - name: install-step
+      image: {{ template "spire-lib.image" (dict "image" .Values.tests.toolkit.image "global" .Values.global) }}
+      env:
+      - name: STEP_VERSION
+        value: 0.25.1
+      workingDir: /data
+      command:
+        - sh
+        - -c
+        - |
+          export URL="https://dl.smallstep.com/gh-release/cli/gh-release-header/v${STEP_VERSION}/step_linux_${STEP_VERSION}_amd64.tar.gz"
+          wget -c $URL -O - | tar --strip-components 2 -C . -xzv "step_${STEP_VERSION}/bin/step"
+          chmod +x /data/step
+          /data/step >/dev/null
+      {{- with .Values.securityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      volumeMounts:
+        - name: data-volume
+          mountPath: /data
```

Comment:
Looks like it worked for testing. ❤️

For a solution though, things shouldn't downloaded at runtime...

I think the "smallstep/step-cli:0.25.1" image might do the trick?

no jq command in it but could still use the toolkit image to do the jq work.

Reply:
I'm fine with that. I'll copy the binary out to `/data` and use it in toolkit image below

Reply:
hmm... that would work, but it may be even simpler to split the last line up:
```
echo $JWT | /data/step crypto jwt verify --key=public.pem --alg=RS256 --subtle
```

to
```
echo $JWT > /data/jwt
```

and then read it in, in an instance of the step container that does
```
cat /data/jwt | /data/step crypto jwt verify --key=public.pem --alg=RS256 --subtle
```

Then you don't need to copy binaries around.

Reply:
yes. but it would could be less fragile to keep the binaries in their respective containers, as if they stop statically linking the step binary, it would become dependent on the image it was built against. while copying the string from container to container should always work?

Reply:
I didn't see cgo, if it's native to there will not be any shared libraries

Reply:
Just my feeling that its better to not copy binaries around if you don't have to. I'm not going to block over it though.